packages/x-data-grid/package.json

```diff
@@ -48,6 +48,7 @@
   },
   "dependencies": {
     "@babel/runtime": "^7.24.5",
+    "@mui/internal-test-utils": "1.0.0-dev.20240529-082515-213b5e33ab",
```

Comment:
@LukasTy this was added as a dependency, causing consumers to get this, and many others deps, installed.
this addition cause peerDep errors (about babel missing and react@18 not supported) on our end.

Reply:
Thank you for raising this! 🙏 
An issue has also been created for it: https://github.com/mui/mui-x/issues/13317.